bundles/CoreBundle/src/EventListener/Frontend/GlobalTemplateVariablesListener.php

```diff
@@ -75,6 +75,10 @@ public function onKernelController(ControllerEvent $event): void
 
     public function onKernelResponse(ResponseEvent $event): void
     {
+        if (!$this->matchesPimcoreContext($event->getRequest(), PimcoreContextResolver::CONTEXT_DEFAULT)) {
+            return;
+        }
+
         if (count($this->globalsStack)) {
```

Comment:
@jdreesen I think with this check this is covered, right?

Reply:
I wasn't sure how this is called. It seems to be some kind of stack for sub-requests. 

If the Pimcore context can't change in between, we're probably safe. 

If it can change, and the first call is “default” context, we will push to the stack. If the second request is not of “default” context, the `onKernelController` listener will be skipped, but the `onKernelResponse` listener will pop the stack. So when we're back at the first request, the entry is missing, because it got removed too early, right? Thus I thought it might be more safe to check the context in `onKernelResponse`, too.